test/Npgsql.Tests/SlimDataSourceTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+
+namespace Npgsql.Tests;
+
+class SlimDataSourceTests : TestBase
```

Comment:
For the other slim opt-in features, I've placed tests in their respective test classes; for example, JsonTests has As_poco_supported_only_with_SystemTextJson. If we want to do it this way, then we should simply add a Records_supported_only_with_EnableRecords in MiscTypeTests